docs/api-version-4-migration-guide.md

```diff
@@ -0,0 +1,97 @@
+`ISBOMReaderForExternalDocumentReference` | `ISbomReaderForExternalDocumentReference`
+`ISBOMValidator` | `ISbomValidator`
+`IValidatedSBOM` | `IValidatedSbom`
+
+### Classes
+
+Old name (all caps) | New name (Pascal case)
+--- | ---
+`ComponentDetectionToSBOMPackageAdapter` | `ComponentDetectionToSbomPackageAdapter`
+`SBOMApiMetadataProvider` | `SbomApiMetadataProvider`
+`SBOMComponentsWalker` | `SbomComponentsWalker`
+`SBOMFile` | `SbomFile`
+`SBOMFormatExtensions` | `SbomFormatExtensions`
+`SBOMMetadata` | `SbomMetadata`
+`SBOMPackagesProvider` | `SbomPackagesProvider`
+`SBOMReference` | `SbomReference`
+`SBOMRelationship` | `SbomRelationship`
+`SBOMTelemetry` | `SbomTelemetry`
+`SBOMValidationResult` | `SbomValidationResult`
+`SPDXSBOMReaderForExternalDocumentReference` | `SPDXSbomReaderForExternalDocumentReference`
+`ValidatedSBOM` | `ValidatedSbom`
+`ValidatedSBOMFactory` | `ValidatedSbomFactory`
+
+### Methods
+
+Old name (all caps) | New name (Pascal case)
+--- | ---
+`IInternalMetadataProvider.GetSBOMNamespaceUri` | `IInternalMetadataProvider.GetSbomNamespaceUri`
+`IRecorder.RecordSBOMFormat` | `IRecorder.RecordSbomFormat`
+`ISBOMGenerator.GetSupportedSBOMSpecifications` | `ISbomGenerator.GetSupportedSbomSpecifications`
+`ISBOMReaderForExternalDocumentReference.ParseSBOMFile`  | `ISbomReaderForExternalDocumentReference.ParseSbomFile`
+`ISbomRedactor.RedactSBOMAsync` | `ISbomRedactor.RedactSbomAsync`
+`ManifestInfo.ToSBOMSpecification` | `ManifestInfo.ToSbomSpecification`
+`SbomConfigProvider.GetSBOMNamespaceUri` | `SbomConfigProvider.GetSbomNamespaceUri`
+`SbomGenerator.GetSupportedSBOMSpecifications` | `SbomGenerator.GetSupportedSbomSpecifications`
+`SbomRedactor.RedactSBOMAsync` | `SbomRedactor.RedactSbomAsync`
+`SPDXSBOMReaderForExternalDocumentReference.ParseSBOMFile` | `SPDXSbomReaderForExternalDocumentReference.ParseSbomFile`
+`TelemetryRecorder.RecordSBOMFormat` | `TelemetryRecorder.RecordSbomFormat`
+`ValidatedSBOMFactory.CreateValidatedSBOM` | `ValidatedSbomFactory.CreateValidatedSbom`
+
+### Properties and const values
+
+Old name (all caps) | New name (Pascal case)
+--- | ---
+`Events.SBOMGenerationWorkflow` | `Events.SbomGenerationWorkflow`
+`Events.SBOMParseMetadata` | `Events.SbomParseMetadata`
+`Events.SBOMValidationWorkflow` | `Events.SbomValidationWorkflow`
+`FileHashes.SBOMFileHash` | `FileHashes.SbomFileHash`
+`IAssemblyConfig.DefaultSBOMNamespaceBaseUri` | `IAssemblyConfig.DefaultSbomNamespaceBaseUri`
+`MetadataKey.SBOMToolName` | `MetadataKey.SbomToolName`
+`MetadataKey.SBOMToolVersion` | `MetadataKey.SbomToolVersion`
+`SettingsSource.SBOMApi` | `SettingsSource.SbomApi`
+
+We discussed making a similar change with "SPDX", but ultimately decided to leave "SPDX" as all caps when dealing with interfaces and classes. There are many places in the code where variables use either camel casing or Pascal casing, but none of those should impact code that builds upon our API.
+
+## Functional changes
```

Comment:
I thought there were also some ISet generic changes that we made, should that be included here?

Reply:
None of them impacted any of the interfaces from the 3.1 API